src/gui/addnewtorrentdialog.h

```diff
@@ -31,20 +31,23 @@
 
 #include <memory>
 
+#include <QByteArray>
 #include <QDialog>
+#include <QString>
+#include <QSize>
```

Comment:
The member variables (`m_storeDialogSize`...) needs it.
`SettingValue<>` template would need the complete type to instantiate it.
Currently it will compile without error is just a coincidence, i.e. required headers was indirectly included via other headers.

Reply:
> The member variables (`m_storeDialogSize`...) needs it.
> `SettingValue<>` template would need the complete type to instantiate it.

No. `SettingValue` doesn't require it (unlike `CachedSettingValue`).
Here is an example that you can test to see the differences:
```c++
template <typename T>
class Test1
{
public:
    T get();
    void set(const T &val);
};

template <typename T>
class Test2
{
public:
    T get();
    void set(const T &val);
    
private:
    T m_field; // T should be complete type
};

class A; // incomplete type

int main()
{
    Test1<A> t1; // OK; Test1 doesn't require A to be complete type
    Test2<A> t2; // Fails; Test2 does require A to be complete type
    return 0;
}
```

Reply:
reverted.